flink-core/src/main/java/org/apache/flink/api/common/typeutils/CompositeType.java

```diff
@@ -132,7 +133,31 @@ public CompositeType(Class<T> typeClass) {
 		}
 		return getNewComparator(config);
 	}
- 	
+
+	// --------------------------------------------------------------------------------------------
+
+	/**
+	 * Debugging utility to understand the hierarchy of serializers created by the Java API.
+	 */
+	public static <T> String getSerializerTree(TypeInformation<T> ti) {
+		return getSerializerTree(ti, 0);
+	}
+
+	private static <T> String getSerializerTree(TypeInformation<T> ti, int indent) {
+		String ret = "";
+		if(ti instanceof CompositeType) {
+			ret += ti.toString()+"\n";
```

Comment:
Should the ```toString``` method not already print the whole tree? Thus, the information would be redundant.